OT111-49 - Devolver token al usuario luego de registrarse en el sitio

denisdimarco opened the pull request:
[https://alkemy-labs.atlassian.net/browse/OT111-49](https://alkemy-labs.atlassian.net/browse/OT111-49)

- Se agrego jwtTokenUtil para poder generar un jwt en el servicio de registro.
- Se modifico el metodo entity2RegisterResponseDTO de AuthenticationMapper para que soporte el jwt al hacer el mappeo.
- Se agrego el atributo String jwt a RegisterResponse.

denisdimarco committed:
OT111-49 - Modified signupUser method of UserDetailsCustomService

- added jwtTokenUtil to generate JWT string in register
- Modified entity2RegisterResponseDTO method in AuthenticationMapper to support  JWT on build.
- Added String jwt to RegisterResponse

OsvaldoGWP commented on src/main/java/com/alkemy/ong/auth/service/UserDetailsCustomService.java:
```diff
@@ -1,8 +1,11 @@
+import com.alkemy.ong.auth.service.impl.UserAuthServiceImpl;
 import com.alkemy.ong.model.entity.UserEntity;
 import com.alkemy.ong.model.mapper.AuthenticationMapper;
+import com.alkemy.ong.model.request.security.AuthenticationRequest;
```
Imports no utilizados, chequear de ver plugin de intellij que revisa esto automaticamente, se llama actionSave

OsvaldoGWP replied:
link https://plugins.jetbrains.com/plugin/7642-save-actions

OsvaldoGWP commented on src/main/java/com/alkemy/ong/auth/service/UserDetailsCustomService.java:
```diff
@@ -27,20 +31,34 @@
     }
 
+    @Autowired
+    private JwtUtil jwtTokenUtil;
```
aqui en este archivo se esta usando 2 tipos de injeccion de dependencias, elegir uno y normalizarlo, en este archivo optaria por añadirlo al constructor porq estan usando @Lazy

denisdimarco committed:
OT111-49 - Fix Pr #40

Removed unused imports
Changed inyection dependency type for JwtUtil to contructor

denisdimarco committed:
Merge branch 'develop' into feature/OT111-49

denisdimarco committed:
OT111-49 - Fix PR#40

Resolved conflicts with develop
